### PR TITLE
0.2.1 - Misc features and updates to make BM more extendable

### LINKED ISSUE #78: [BUG]: Fix NPM Audit
### Description of the bug

NPM is currently complaining about a number of failed audits. These should be fixed to ensure that the package is properly secured.

### Steps To Reproduce

`npm audit`

### Additional Information

_No response_

### LINKED ISSUE #129: Slotfill: Revisit the `saveByline` implementation
### Description

Review the implementation of how we are saving the profiles in this `useEffect` hook.

https://github.com/alleyinteractive/byline-manager/blob/production/client/src/components/byline-slot/index.jsx#L37-L41

### Use Case

Any site using this plugin.

### LINKED ISSUE #130: Register the labels for the custom taxonomy
### Description

Add custom taxonomy labels here: https://github.com/alleyinteractive/byline-manager/blob/eb1b409f6eabfaa276fa59a191be479108eca83d/inc/data-structures.php#L92

Currently, it falls back to Tags. Plugins making use of this data outputs incorrect, or confusing, data.

Screenshot 2023-07-13 at 14 45 43

### Use Case

A plugin extending Byline Manager.

### LINKED ISSUE #135: Deprecate Buddy pipeline
### Description

As titled.

There is already a Github Action workflow for the built branch. And this pipeline is not used here.

### Use Case

As titled.

### LINKED ISSUE #136: Deprecate `useDebounce` in favor of @uidotdev/usehooks
### Description

As titled.

See https://github.com/alleyinteractive/alley-scripts/issues/250 in case it is bundled into `@alleyinteractive/block-editor-tools`

### Use Case

As titled.

### LINKED ISSUE #137: Add a Slotfill Provider for BM
### Description

As titled.

See https://developer.wordpress.org/block-editor/reference-guides/components/slot-fill/

### Use Case

A new `BylineSlotContainer` container (being worked on) can be filled below the `BylineSlot` slotfill instead of a completely separate slotfill.

### LINKED ISSUE #139: Update the redux store and organize it in the WordPress way
### Description

If you see the redux store organization of one of the blocks in the block editor, it is not similar to the one used in this project. Which adds more "weight" to understand how the project is setup and customize.

### Use Case

Makes it easier for a developer to understand how the redux store is organized.

### PR DESCRIPTION
**Pull request by renatonascalves**
TL;DR: The changes here are to make it easier to extend the Byline Components for third party developers.

The main use case for this feature is having custom slotfills with support for the Byline Manager components (`BylineAutocomplete`, `BylineFreeform`, etc), allowing role management by using a custom meta key instead of a custom taxonomy. See #8 for a different approach.

The block editor components were updated, and in some cases refactored for reusability. It is technically possible now to create new slotfills that saves to different meta keys but using Byline Manager functionality.

**Sidenote**: due to the differences in the schema between the redux store and post meta, when creating a custom slotfill, a new redux store needs to be created, besides passing the meta key. You can handle all that yourself, if desired. The components are flexible enough. But I found it to be a bit of a hassle since the components expects the data in a specific shape.

---

Here is an example of a custom slotfill that uses the new Byline Manager Slotfill provider. This new slotfill will be located below the core section.

```jsx
import { registerPlugin } from '@wordpress/plugins';
import { storeCreator } from 'path/to/byline/manager/store';
import { BylineManagerPanelInfo } from 'path/to/byline/byline-panel-info-container';
import BylineSlotContainer from 'path/to/byline/byline-slot-container';

const myCustomReduxStore = 'byline-contributors';
const myCustomMetaKey = 'byline-contributors';

const MyCustomSlotFill = () => (
  <BylineManagerPanelInfo>
    <p>
      <strong>{__('My Custom Byline Section')}</strong>
    </p>
    <BylineSlotContainer store={bylineManagerStore} metakey={myCustomMetaKey} />
  </BylineManagerPanelInfo>
);

storeCreator(myCustomReduxStore, myCustomMetaKey);
registerPlugin('my-slotfill', { render: MyCustomSlotFill });
```

The meta key for this slotfill needs to be created to be available in the REST API. A new helper function is available for the schema.

```php
register_post_meta(
	'',
	'byline-contributors',
	[
		'single'       => true,
		'type'         => 'object',
		'show_in_rest' => [
			'schema' => \Byline_Manager\byline_meta_schema(),
		],
	]
);
```
---

https://github.com/alleyinteractive/byline-manager/assets/19148962/bd41e7ca-b389-4dbb-9989-aa31ee388850

---

- [x] #78 
- [x] #129
- [x] #130
- [x] #135 
- [x] #136 
- [x] #137 
- [x] #139 
- [x] Added: PHPStan support (not via the CI yet since there a few issues to resolve) See #141
- [x] Updated: Github Actions;
- [x] Updated: Tested up to 6.3;
- [x] Updated: Remove deprecated `Button` props;
- [ ] Add an `examples` folder with custom slotfills samples;

fixes #78
fixes #129
fixes #130
fixes #135
fixes #136
fixes #137
fixes #139

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
   // Do NOT change these rules
   "rules": {
     "indent": [2, 2, {"SwitchCase": 1}],
-    "max-len": [2, 80, 4, {
+    "max-len": [2, 100, 4, {
```
Comment (kevinfodness) [inline .eslintrc.json:41]: Not today's problem, but we should update this package to use our eslint rules from alley-scripts

File: byline-manager.php [hunk 2]
```diff
@@ -6,9 +6,9 @@
  * Author:          Alley Interactive
  * Text Domain:     byline-manager
  * Domain Path:     /languages
- * Version:         0.2.0
+ * Version:         0.2.1
  * Requires PHP:    7.4
- * Tested up to:    4.9
+ * Tested up to:    6.2.2
```
Comment (kevinfodness) [inline byline-manager.php]: Since 6.3 is out now, can we update it to 6.3 instead? Did you test it on 6.3 locally?

File: client/src/components/byline-slot-wrapper/index.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  addFreeformButtonLabel,
+  profilesApiUrl,
+}) => (
+  <div className="components-base-control">
+    {null === profiles ? (
+      <div style={{ textAlign: 'center' }}>
+        <Spinner />
+      </div>
+    ) : (
+      <Fragment>
```
Comment (kevinfodness) [inline client/src/components/byline-slot-wrapper/index.jsx:33]: Should be able to use `<>` instead of `<Fragment>`

**Review by kevinfodness — APPROVED** [review]
🍣

File: .eslintrc.json (hunk 1 above)
Comment (renatonascalves) [reply .eslintrc.json:41]: It's on my list of TODO.

File: byline-manager.php (hunk 2 above)
Comment (renatonascalves) [reply byline-manager.php]: Not yet, I had tested it with 6.2.2.  Will test with 6.3 before merging it.

File: client/src/components/byline-slot-wrapper/index.jsx (hunk 3 above)
Comment (renatonascalves) [reply client/src/components/byline-slot-wrapper/index.jsx:33]: It's currently not possible due to the current eslint config.
